### ISSUE TITLE
请问SMOKE 可以把backbone 换成resnet吗

### ISSUE THREAD

**Issue by stevenk12** [PaddlePaddle/Paddle3D#92]

**Comment by nepeplwu** [comment]
@stevenk12 可以的，需要在配置文件中做如下的修改：
1. train_dataset.transforms.Gt2SmokeTarget 设置 output_stride字段，设置为实际值
2. val_dataset.transforms.Gt2SmokeTarget 设置 output_stride字段，设置为实际值
3. model.head.in_channels 设置为resnet的实际output_channel

**Comment by nepeplwu** [comment]
但是使用resnet不一定速度会快或者效果会好

**Comment by stevenk12** [comment]
> 但是使用resnet不一定速度会快或者效果会好

好的，非常感谢。我昨天已经改了output_channel和output_stride，但还是报错了如下：

Error: /data/jenkins_workspace/workspace/paddle_release/paddle/fluid/operators/gather.cu.h:62 Assertionindex_value >= 0 && index_value < input_dims[i] failed. The index is out of bounds, please check whether the dimensions of index and input meet the requirements . It should be less than [7680] and greater than or equal to 0, but received [8606]

**Comment by nepeplwu** [comment]
修改后的配置文件麻烦发下

**Comment by stevenk12** [comment]
”smoke_dla34_no_dcn_kitti.yml“：

_base_: '../_base_/kitti_mono.yml'

batch_size: 8
iters: 70000

train_dataset:
  transforms:
    - type: LoadImage
      reader: pillow
      to_chw: False
    - type: Gt2SmokeTarget
      mode: train
      num_classes: 9
      input_size: [1280, 384]  
    - type: Normalize
      mean: [0.485, 0.456, 0.406]
      std: [0.229, 0.224, 0.225]

val_dataset:
  transforms:
    - type: LoadImage
      reader: pillow
      to_chw: False
    - type: Gt2SmokeTarget
      mode: val
      num_classes: 9
      input_size: [1280, 384]
    - type: Normalize
      mean: [0.485, 0.456, 0.406]
      std: [0.229, 0.224, 0.225]

optimizer:
  type: Adam

lr_scheduler:
  type: MultiStepDecay
  milestones: [36000, 55000]
  learning_rate: 1.25e-4 

model:
  type: SMOKE
  backbone:
    type: ResNet  
    # This will automatically save to ~/.paddle3d/pretrained/dla34/dla34.pdparams
    # pretrained: "https://bj.bcebos.com/paddle3d/pretrained/dla34.pdparams"
  head:
    type: SMOKEPredictor
    num_classes: 9
    reg_channels: [1, 2, 3, 2, 2]
    num_chanels: 256
    norm_type: "gn"
    in_channels: 2048  
  depth_ref: [70.10, 34.82]
  dim_ref: [[4.26, 1.28, 1.68], [4.64, 1.70, 1.76], [6.33, 2.70, 2.49], [11.00, 3.03, 2.60], [0.48, 1.61, 0.49], [1.51, 1.47, 0.48], [1.67, 1.42, 0.55], [1.0, 1.1, 0.6], [2.74, 1.74, 1.16]]
  max_detection: 50
  pred_2d: True



“../paddle3d/transforms/target_generator.py”：
...
class Gt2SmokeTarget(TransformABC):
    def __init__(self,
                 mode: str,
                 num_classes: int,
                 flip_prob: float = 0.5,
                 aug_prob: float = 0.3,
                 max_objs: int = 50,
                 input_size: Tuple[int, int] = (1280, 384),
                 output_stride: Tuple[int, int] = (8, 8), 
                 # output_stride: Tuple[int, int] = (4, 4),
                 shift_range: Tuple[float, float, float] = (),
                 scale_range: Tuple[float, float, float] = ()):
...

**Comment by stevenk12** [comment]
> 修改后的配置文件麻烦发下

问题解决了 谢谢谢谢
